### DISCUSSION TITLE
Command line interface

### DISCUSSION THREAD

**Discussion by lgrcia** [lgrcia/prose#114] [Ideas]
Hi all! I am thinking about making some CLI tools around *prose*, so we don't have to open notebooks to do regular checks and common reductions.

I already wrote some around the `FitsManager`
- to parse folders: `fitsmanager parse folder_path -f db.sqlite`
- to explore *sqlite* files: `fitsmanager observations db.sqlite`

So few questions:
- What kind of tools do we want?
- Should this live in another package?
- Who can help?

Please feel free to share ideas in whatever format and details you like!

Cheers!

**Comment by alexdumitriu** [comment]
Hi Lionel,

From my workflow, I was thinking about the following:

- A way to export a target's lightcurve in some csv/txt format that can be ingested by AAVSO or ETD for reporting purposes.
- Detrending fluxes acquired over a few separate nights. I was looking at [PDTrend](https://github.com/dwkim78/pdtrend), but not sure how realistic that is.
- And one absolute wishlist item: phase plotting function for lightcurves

Let me know if there's anything I can do to help!

Cheers,
Alex

**Reply by lgrcia** [reply]
Hi @alexdumitriu,

I think that these are great additional features for the `Fluxes` object. Although not very exploited, this object can already be saved and exported to csv/txt quite easily, but from a python script. So I understand you would like to have command line tools to interact with it, right? (or a tutorial?)

As for the detrending I think there are options, but these might rather come from another tool/package.

Thanks a lot for these! I just opened issue #121 to follow on some development ideas.

**Reply by alexdumitriu** [reply]
Hi @lgrcia,

Yes, I was thinking from command line. After saving the fluxes, it would be good to have a CLI tool to interact with the individual targets fluxes in a way like this `fluxes -input=all_fluxes_files -target=2 -export=csv/txt`

Cheers,
Alex

**Comment by mathvdd** [comment]
Hello Lionel,

I was thinking about a CLI to check the centering of the objects in my pipeline. As we discussed already combination of long exposures, narrow-band images and extended objects makes it challenging to find the optocenter of the object in an image. I am thinking about an efficient way to visualize the result of the centering and rerun centering algorithm if needed.

Not sure exactly how much of this is possible with a CLI but the goal would be the following:
- the image or a part of the image would be displayed with the result of the centering. Different contrast algorithms would be interesting as the objects can be quite faint and close to the noise limit.
- the centering algorithm could be rerun starting form user input coordinates, for example clicking on the images or navigating the image using the keyboard
- upon satisfactory results the next image is loaded or another part of the pipeline is called

I am still investigating the feasibility of doing that directly in the terminal, but if you are interested we can discuss it :)

Cheers,
Mathieu
